app/forms/search_developer_form.rb

```diff
@@ -9,8 +9,6 @@ class SearchDeveloperForm
 
   def search
     query = Developer.all
-    return query if !programming_language_id? && !language_id?
```

Comment:
Then you should delete the unused methods (`programming_language_id?` and `language_id?`), too.